src/Input/LRealInput.st

```diff
@@ -54,5 +59,16 @@ NAMESPACE Simatic.Ax.IO.Input
                 Q := _default;
             END_IF;     
         END_METHOD
+
+        /// Returns TRUE if current value is not equal to last value
+        /// within a tolerance
+        METHOD PUBLIC HasChanged : BOOL
```

Comment:
Proposal: does it make sense for analog values to use a hysteresis for the HasChanged() method?

Reply:
That depends on how you mean, the tolerance as I named it here is a sort of hysteresis but it moves with _last_value to prevent HasChanged to capture the jitter. It might be nice to add a separate functionality though that compares against a set value together with an optional hysteresis. That said it would probably be useful to implement it for all other analog types too.

I do not know if this is a real concern for the modern CPUs, but it might be good to split the IO classes into tiers of functionality. If you have a lot of analog inputs the memory usage will be much higher if all of them have complete functionality, which will be wasted memory if not all use the functionalities.

Reply:
The reason for this was for us to easily trigger actions when the value has changed "by atleast X" amount, sort of like a pos/neg flank for signals. In our application we are not concerned with shorter peaks. If this is not something that is commonly used then maybe this would fit better in a separate library in our organization? :)

Reply:
I discussed within the team. So, it's fine to choose your way. If somebody requests a hysteresis, we can reopen the discussion.